pkg/worker/task_runner.go

```diff
@@ -17,75 +19,123 @@ import (
 
 type TaskRunner struct {
 	conductorTaskResourceClient *conductor_http_client.TaskResourceApiService
-	waitGroup                   sync.WaitGroup
-}
+	hostName                    string
 
-func NewWorkerOrkestratorWithApiClient(
-	apiClient *conductor_http_client.APIClient,
-) *TaskRunner {
-	return &TaskRunner{
-		conductorTaskResourceClient: &conductor_http_client.TaskResourceApiService{
-			APIClient: apiClient,
-		},
-	}
+	maxAllowedWorkersByTaskType map[string]int
+	runningWorkersByTaskType    map[string]int
+	mutex                       sync.Mutex
+	workerWaitGroup             sync.WaitGroup
 }
 
-func NewTaskRunner(
-	authenticationSettings *settings.AuthenticationSettings,
-	httpSettings *settings.HttpSettings,
-) *TaskRunner {
+func NewTaskRunner(authenticationSettings *settings.AuthenticationSettings, httpSettings *settings.HttpSettings) *TaskRunner {
 	apiClient := conductor_http_client.NewAPIClient(
 		authenticationSettings,
 		httpSettings,
 	)
+	return NewTaskRunnerWithApiClient(apiClient)
+}
+
+func NewTaskRunnerWithApiClient(
+	apiClient *conductor_http_client.APIClient,
+) *TaskRunner {
+	hostname, err := os.Hostname()
+	if err != nil {
+		hostname = ""
```

Comment:
not sure if it would cause problems on the server side or not. so in other words, should we error out here?

Reply:
This is just for the informational purpose, so if we cannot get hostname, its okay.